version: '3.8'

services:
  database:
    image: postgres
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - network
    container_name: django-db
    restart: always
    env_file: .env
    healthcheck:
      test: "pg_isready -U $POSTGRES_USER"
      interval: 5s
      timeout: 5s
      retries: 5
 
django-web:
  build: .
  container_name: django-docker
  ports:
    - "8000:8000"
  depends_on:
    - database
  networks:
    - network
  environment:
    DJANGO_SECRET_KEY: ${DJANGO_SECRET_KEY}
    DEBUG: ${DEBUG}
    DJANGO_LOGLEVEL: ${DJANGO_LOGLEVEL}
    DJANGO_ALLOWED_HOSTS: ${DJANGO_ALLOWED_HOSTS}
    DATABASE_ENGINE: ${DATABASE_ENGINE}
    DATABASE_NAME: ${DATABASE_NAME}
    DATABASE_USERNAME: ${DATABASE_USERNAME}
    DATABASE_PASSWORD: ${DATABASE_PASSWORD}
    DATABASE_HOST: ${DATABASE_HOST}
    DATABASE_PORT: ${DATABASE_PORT}
  env_file:
    - .env

volumes:
   postgres_data:

networks:
  network:
    driver: bridge